### PR TITLE
RUST-1509 SDAM Logging

### PR DESCRIPTION
**Pull request by isabelatkinson**

### CODE REVIEW

**Review by abr-egn — COMMENTED** [review]
Mostly LGTM!  Just a few minor comments.

File: src/sdam/topology.rs [hunk 1]
```diff
@@ -67,22 +70,38 @@ pub(crate) struct Topology {
...
-        });
+        let event_emitter =
+            if options.sdam_event_handler.is_some() || cfg!(feature = "tracing-unstable") {
+                let user_handler = options.sdam_event_handler.clone();
+
+                #[cfg(feature = "tracing-unstable")]
+                let tracing_emitter =
+                    TopologyTracingEventEmitter::new(options.tracing_max_document_length_bytes, id);
+                let (tx, mut rx) = mpsc::unbounded_channel::<AcknowledgedMessage<SdamEvent>>();
+                // Spin up a task to handle events so that a user's event handling code can't block
```
[the remark this answers is not in the document]
Comment (isabelatkinson) [reply src/sdam/topology.rs:83]: good catch, I'll just remove this comment.

File: src/test/spec.rs [hunk 2]
```diff
@@ -71,6 +71,12 @@ pub(crate) fn deserialize_spec_tests<T: DeserializeOwned>(
             continue;
         };
 
+        if let Ok(unskipped_filename) = std::env::var("TEST_FILE") {
```
[the remark this answers is not in the document]
Comment (isabelatkinson) [reply src/test/spec.rs:74]: I add this when I'm debugging pretty often to make it easier to run single tests, but I can delete if you'd rather not have it checked in

[the remark this answers is not in the document]
Comment (abr-egn) [reply src/test/spec.rs:74]: Nah, it seems fine to leave in as long as it's deliberate :)

**Review by abr-egn — APPROVED** [review]

**Pull request merged**